src/RSocketClient.cpp

```diff
@@ -60,6 +63,30 @@ folly::Future<std::unique_ptr<RSocketRequester>> RSocketClient::connect(
   return future;
 }
 
+void RSocketClient::resume(
+    std::unique_ptr<ClientResumeStatusCallback> resumeCallback) {
+  CHECK(stateMachine_);
+  connectionFactory_->connect([
+    this,
+    resumeCallback = std::move(resumeCallback)
+  ](std::unique_ptr<DuplexConnection> connection,
+    folly::EventBase & evb) mutable {
+    CHECK(evb_ == &evb);
+    auto frameTransport =
+        std::make_shared<FrameTransport>(std::move(connection));
+    stateMachine_->tryClientResume(
+        resumeToken_, std::move(frameTransport), std::move(resumeCallback));
+  });
+}
+
+void RSocketClient::disconnect(folly::exception_wrapper ex) {
```

Comment:
If this is going to move here, then `RSocketRequester` should not have `closeSocket`: https://github.com/rsocket/rsocket-cpp/blob/master/src/RSocketRequester.h#L96